### DISCUSSION TITLE
make resolved pathname available on router object

### DISCUSSION THREAD

**Discussion by stefanprobst** [vercel/next.js#23966] [Ideas]
# Feature request

## Is your feature request related to a problem? Please describe.

when setting `aria-current` on `NavLink`s it is helpful to get everything necessary for the matching function from `useRouter`. for dynamic routes, `useRouter` currently returns the `route` (name of the file in `pages` folder), a `query` object, and `asPath` which includes pathname+query+hash.

however, the resolved pathname of the current route is not directly available from `useRouter`.

## Describe the solution you'd like

`useRouter` should return an additional property (named `resolvedPathname` or similar). 
for example on a `/posts/[id]` route, when navigating to `/posts/one?category=new`, `useRouter` would then return:
```js
{
  "pathname": "/posts/[id]",
  "route": "/posts/[id]",
  "query": {
    "id": "one",
    "category": "new"
  },
  "asPath": "/posts/one?category=new",
  "resolvedPathname": "/posts/one" // this would be added
}
```

## Describe alternatives you've considered

use `new URL(router.asPath).pathname` or `window.location.pathname`

**Comment by balazsorban44** [comment]
This is something I constantly come across myself! I end up using the following everywhere, so I created a hook, but it would be convenient to get it from `useRouter()`:

```js
const router = useRouter()
  return router.asPath
    .replace(/#.*/, "")
    .replace(/\?.*/, "")
```
(I might rewrite it to your workaround for now)

@stefanprobst `new URL(router.asPath).pathname` would fail though:
![image](https://user-images.githubusercontent.com/18369201/95828568-6e4e8700-0d35-11eb-8bc8-c30af7f8a3dc.png)

You could overcome this with a fake (but valid) basepath, like `new URL(router.asPath, "http://n").pathname`

**Comment by stefanprobst** [comment]
> You could overcome this with a fake (but valid) basepath, like `new URL(router.asPath, "http://n").pathname`

oops, thanks - this is exactly what i do currently!
